File: src/main/java/org/qxdn/birthdayreminder/model/model/Character.java
package org.qxdn.birthdayreminder.model.model;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.qxdn.birthdayreminder.model.model.content.CharacterContent;

import java.util.ArrayList;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
public class Character extends BaseModel {

    private String name;

    private String originName;

    private List<String> otherName = new ArrayList<>();

    /**
     * 角色性别 考虑到性别秀吉，还是直接中文
     */
    private String gender;

    private Integer birthYear;

    private Integer birthMonth;

    private Integer birthDay;

    /**
     * bangumi的id
     */
    private Long bangumiId;

    private CharacterContent content = new CharacterContent();

    private String comment;
}

File: model/model/character.go
package model

type Character struct {
	BaseModel
	Name       string           // 名称
	OriginName string           // 原名
	OtherName  []string         // 其他名字
	BirthYear  int              // 出生年
	BirthMonth int              // 出生月
	BirthDay   int              // 出生日
	content    CharacterContent // 拓展内容
	Comment    string           // 备注
}
